src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -175,11 +175,18 @@ public Image<TPixel> Decode<TPixel>(Stream stream)
                                     this.InitializeImage(metadata, out image);
                                 }
 
-                                using (var deframeStream = new ZlibInflateStream(this.currentStream, this.ReadNextDataChunk))
+                                var deframeStream = new ZlibInflateStream(this.currentStream, this.ReadNextDataChunk);
```

Comment:
I think keeping the `using` block would be cleaner here. Or am I missing something?

Reply:
Yeah, thought it was throwing after we assign the stream but it isn't so have refactored.